src/adservice/src/main/java/hipstershop/AdService.java

```diff
@@ -91,8 +100,14 @@ private static class AdServiceImpl extends hipstershop.AdServiceGrpc.AdServiceIm
     @Override
     public void getAds(AdRequest req, StreamObserver<AdResponse> responseObserver) {
       AdService service = AdService.getInstance();
+
+      // get the current span in context
+      Span span = Span.current();
       try {
         List<Ad> allAds = new ArrayList<>();
+
+        span.setAttribute("app.ads.contextKeys", req.getContextKeysList().toString());
+        span.setAttribute("app.ads.contextKeys.count", req.getContextKeysCount());
```

Comment:
Should we use `app.ads` or `app.adservice` as the namespace

Reply:
I would argue against including the `service` suffix in all attribute names.

Reply:
Doing instrumentation from other services, we are using attribute names that don't reflect the service they are defined in.  Such as `app.order.id` which is part of the email service. If we are to use the service names in the attribute names, this should be handled in a larger conversation.

Reply:
What would the intent be behind adding service name? If it's part of a span in a service it's inherently part of that service unless that's not captured in the span info? If it's for querying I feel like adding a single service name attribute is preferable to every attribute having a service prefix

Reply:
I don't think we need to namespace the service name in these attribute keys? There's already service.name.